### PR TITLE
Fix #1179 : Impersonate-Extra keys may be specified multiple times

### LINKED ISSUE #1179: Impersonate-Extra keys may be specified multiple times
According to https://kubernetes.io/docs/reference/access-authn-authz/authentication/#user-impersonation, the `Impersonate-Extra` HTTP header can be specified multiple times, however `RequestConfig.setImpersonatorExtra` only accepts a single value for each entry.

### PR DESCRIPTION
**Pull request by rohanKanojia**
#1179

### CODE REVIEW

File: kubernetes-client/src/main/java/io/fabric8/kubernetes/client/utils/ImpersonatorInterceptor.java [hunk 1]
```diff
@@ -46,8 +47,8 @@ public Response intercept(Chain chain) throws IOException {
       }
 
       if (isNotNullOrEmpty(requestConfig.getImpersonateExtras())) {
-        for (Map.Entry<String, String> entry : requestConfig.getImpersonateExtras().entrySet()) {
-          requestBuilder.addHeader("Impersonate-Extra-" + entry.getKey(), entry.getValue());
+        for (Map.Entry<String, List<String>> entry : requestConfig.getImpersonateExtras().entrySet()) {
+          requestBuilder.addHeader("Impersonate-Extra-" + entry.getKey(), String.join(",", entry.getValue()));
```
[the remark this answers is not in the document]
Comment (rohanKanojia) [reply kubernetes-client/src/main/java/io/fabric8/kubernetes/client/utils/ImpersonatorInterceptor.java]: Actually I saw in Kubernetes repo([impersonation.go](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/endpoints/filters/impersonation.go#L69)), and changed impersonate extra to `Map<String, List<String> >` from `Map<String, String>` . I added this join clause to handle cases like these: 
```
Impersonate-Extra-dn: cn=jane,ou=engineers,dc=example,dc=com
Impersonate-Extra-acme.com%2Fproject: some-project
Impersonate-Extra-scopes: view
Impersonate-Extra-scopes: development
```
You mean I should add header entry for each list element like this:
```
Impersonate-Extra-dn: cn=jane
Impersonate-Extra-dn: ou=engineers
Impersonate-Extra-dn: dc=example
Impersonate-Extra-dn: dc=com
Impersonate-Extra-acme.com%2Fproject: some-project
Impersonate-Extra-scopes: view
Impersonate-Extra-scopes: development
```

[the remark this answers is not in the document]
Comment (gastaldi) [reply kubernetes-client/src/main/java/io/fabric8/kubernetes/client/utils/ImpersonatorInterceptor.java]: Yes, the spec says that the same `Impersonate-Extra-*` header can appear multiple times in the request

[the remark this answers is not in the document]
Comment (rohanKanojia) [reply kubernetes-client/src/main/java/io/fabric8/kubernetes/client/utils/ImpersonatorInterceptor.java]: okay, Thanks for clarifying ;-) . Will do update accordingly

File: kubernetes-client/src/main/java/io/fabric8/kubernetes/client/utils/ImpersonatorInterceptor.java [hunk 2]
```diff
@@ -45,9 +48,14 @@ public Response intercept(Chain chain) throws IOException {
...
       }
 
-      if (isNotNullOrEmpty(requestConfig.getImpersonateExtras())) {
-        for (Map.Entry<String, String> entry : requestConfig.getImpersonateExtras().entrySet()) {
-          requestBuilder.addHeader("Impersonate-Extra-" + entry.getKey(), entry.getValue());
+      Map<String, List<String>> impersonateExtras = requestConfig.getImpersonateExtras();
+      if (isNotNullOrEmpty(impersonateExtras)) {
+        Collection<?> keys = impersonateExtras.keySet();
+        for (Object key : keys) {
+          List<String> values = impersonateExtras.get(key);
```
Comment (gastaldi) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/utils/ImpersonatorInterceptor.java]: What if values is null?

**Comment by gastaldi** [comment]
[merge]

**Comment by oscerd** [comment]
[merge]

**Pull request merged**